### DISCUSSION TITLE
running flask db from within custom package

### DISCUSSION THREAD

**Discussion by amouhsine** [nickjj/flask-db#3] [Q&A] [answered]
Hi @nickjj ,

First of all, I would like to congratulate you on flask-db. ( I am also keen to invest in your snakeeyes app!)
I  am writing a flask app and I don't want to let the user write the `export FLASKAPP=` and then `flask db` to avoid messing with the environment variables. 
What is the best way to access the db cli via a console script defined in the setup.py ?
Basically, I want something like `manage db --help` where manage points to manage.py. In this file, I will handle all the environments variables.

Thanks in advance for your help.

A

**Comment by nickjj** [comment, marked as answer]
Hi,

Setting `FLASK_APP` is a requirement of the `flask` CLI command not this extension.

If you don't set that variable then even running `flask --help` will throw an error of `flask.cli.NoAppException: Could not locate a Flask application. You did not provide the "FLASK_APP" environment variable, and a "wsgi.py" or "app.py" module was not found in the current directory.`.

You can choose to use a `.env` file and have that loaded in with `python-dotenv` (or Docker instead if your app is using it). If you're using systemd you can load in an env file too. This way the user never has to export anything directly. The same goes for any cloud environment like Heroku too.
